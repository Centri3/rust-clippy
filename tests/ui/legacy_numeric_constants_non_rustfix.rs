//@aux-build:proc_macros.rs:proc-macro
#![allow(clippy::no_effect, deprecated, unused)]
#![warn(clippy::legacy_numeric_constants)]

#[macro_use]
extern crate proc_macros;

use std::u128 as _;

fn main() {
    use std::u32;
    u32::MAX;
    use std::f64;
    f64::MAX;
    u128::MAX;
    // Don't lint
    u8::MIN;
    std::f32::consts::E;
    use std::f32;
    f64::consts::E;
}
