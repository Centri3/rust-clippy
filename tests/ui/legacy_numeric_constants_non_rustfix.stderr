error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants_non_rustfix.rs:12:5
   |
LL |     u32::MAX;
   |     ^^^^^^^^ help: use the associated constant instead: `u32::MAX`
   |
note: you may need to remove one of the following `use` statements
  --> $DIR/legacy_numeric_constants_non_rustfix.rs:11:5
   |
LL |     use std::u32;
   |     ^^^^^^^^^^^^^
   = note: `-D clippy::legacy-numeric-constants` implied by `-D warnings`

error: usage of a legacy numeric constant
  --> $DIR/legacy_numeric_constants_non_rustfix.rs:14:5
   |
LL |     f64::MAX;
   |     ^^^^^^^^ help: use the associated constant instead: `f64::MAX`
   |
note: you may need to remove one of the following `use` statements
  --> $DIR/legacy_numeric_constants_non_rustfix.rs:13:5
   |
LL |     use std::f64;
   |     ^^^^^^^^^^^^^

error: aborting due to 2 previous errors

